akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -283,6 +284,39 @@ final case class Intersperse[T](start: Option[T], inject: T, end: Option[T]) ext
 /**
  * INTERNAL API
  */
+private[akka] final case class Dedupe[T]() extends GraphStage[FlowShape[T, T]] {
+  private val in = Inlet[T]("Dedupe.in")
+  private val out = Outlet[T]("Dedupe.out")
+
+  override def shape: FlowShape[T, T] = FlowShape(in, out)
+  override def initialAttributes = DefaultAttributes.dedupe
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    var last: Any = null
+    val inHandler = new InHandler {
```

Comment:
No need to increase the size of the GraphStageLogic class by having these as fields, use setHandler directly to the new InHandler/OutHandler.